client.go

```diff
@@ -292,10 +292,13 @@ func (c *Client) wrapCall(client *core.Client, ctx context.Context, serviceMetho
 
 //Go invokes the function asynchronously. It returns the Call structure representing the invocation.
 //The done channel will signal when the call is complete by returning the same Call object. If done is nil, Go will allocate a new channel. If non-nil, done must be buffered or Go will deliberately crash.
-func (c *Client) Go(ctx context.Context, serviceMethod string, args interface{}, reply interface{}, done chan *core.Call) *core.Call {
-	rpcClient, _ := c.ClientSelector.Select(c.ClientCodecFunc)
+func (c *Client) Go(ctx context.Context, serviceMethod string, args interface{}, reply interface{}, done chan *core.Call) (*core.Call, error) {
```

Comment:
There is error information that is being lost here with the _ underscore, if the connection is mis-configured (this bit me when trying to figure out the kcp use). If the error is not returned, the alternative is to panic... which is also undesirable in production, but panic is probably equivalent to what happens now, only it would come earlier so would offer better insight into the problem. I'll change it to panic.